code/data_structures/src/queue/priority_queue/priority_queue.cpp

```diff
@@ -0,0 +1,115 @@
+// C++ implementation of Priority Queue
+
+#include <iostream>
+#include <cstdio>
+#include <cstring>
+#include <cstdlib>
+ 
+// Node Declaration
+struct node {
+	int priority;
+	int info;
+	node *link;
+};
+
+class PriorityQueue {
+private:
+    node *front;
+public:
+    PriorityQueue() {
+        front = nullptr;
+    }
+
+    // Add an element into Priority Queue according to its priority
+    void add(int item, int priority) {
+        node *tmp, *q;
+        tmp = new node;
+        tmp->info = item;
+        tmp->priority = priority;
+        if (front == nullptr || priority < front->priority){
+            tmp -> link = front;
+            front = tmp;
+        }
+        else {
+            q = front;
+            while (q->link != nullptr && q->link -> priority <= priority)
+                q = q->link;
+
+            tmp->link = q->link;
+            q->link = tmp;
+        }
+    }
+    
+    // Removes an element with highest priority from Priority Queue 
+    void remove() {
+        node *tmp;
+        if(front == nullptr)
+            throw true;
+        else {
+            tmp = front;
+            std::cout << "Removed item is: " << tmp->info << "\n";
```

Comment:
Remove this.

Reply:
I am actually not clear why should I remove this ??